File: src/main/java/br/com/jiratorio/config/security/SignerConfiguration.java
package br.com.jiratorio.config.security;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.jwt.crypto.sign.MacSigner;

@Configuration
public class SignerConfiguration {

    @Bean
    public MacSigner signer(@Value("security.key") final String key) {
        return new MacSigner(key);
    }

}

File: src/main/kotlin/br/com/jiratorio/config/security/SignerConfiguration.kt
package br.com.jiratorio.config.security

import br.com.jiratorio.config.properties.SecurityProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.jwt.crypto.sign.MacSigner

@Configuration
class SignerConfiguration {

    @Bean
    fun signer(securityProperties: SecurityProperties) =
        MacSigner(securityProperties.key)

}
